File: 2225-find-players-with-zero-or-one-losses/2225-find-players-with-zero-or-one-losses.cs
public class Solution {
    public IList<IList<int>> FindWinners(int[][] matches) {
        List<IList<int>> ret = new List<IList<int>>();

        SortedDictionary<int, int> dic = new SortedDictionary<int, int>();
        
        foreach(var match in matches){
            dic[match[0]] = dic.GetValueOrDefault(match[0], 0);
            dic[match[1]] = dic.GetValueOrDefault(match[1], 0) + 1;
        }
        
        List<int> retList1 = new List<int>();
        List<int> retList2 = new List<int>();
        
        foreach(var d in dic){
            if(d.Value == 0) retList1.Add(d.Key);
            else if(d.Value == 1) retList2.Add(d.Key);
        }
        ret.Add(retList1);
        ret.Add(retList2);
            
        return ret;
    }    
}

File: 2225-find-players-with-zero-or-one-losses/2225-find-players-with-zero-or-one-losses.ts
function findWinners(matches: number[][]): number[][] {
    const map = new Map<number, number>();
    const ret1: number[] = [];
    const ret2: number[] = [];    
    
    for(let m of matches){
        
        map.set(m[0], (map.get(m[0]) || 0));
        map.set(m[1], (map.get(m[1]) || 0) + 1);
    }
    
    for(const [key, value] of map){
        if(value === 0) ret1.push(key);
        else if(value === 1) ret2.push(key);
    }
    
    ret1.sort((a,b)=>a-b);
    ret2.sort((a,b)=>a-b);
    
    return [ret1, ret2];
};